File: leetcode_2696/MinimumStringLengthAfterRemovingSubstrings.java
package leetcode_2696;

import java.util.Stack;

class MinimumStringLengthAfterRemovingSubstrings {
    public int minLength(String s)
    {
        Stack<Character> stack = new Stack<>();
        for(char c : s.toCharArray())
        {
            if(stack.isEmpty())
            {
                stack.push(c);
                continue;
            }
            if(c == 'B' && stack.peek() == 'A')
                stack.pop();
            else if(c == 'D' && stack.peek() == 'C')
                stack.pop();
            else
                stack.push(c);
        }
        return stack.size();
    }

    public static void main(String[] args) {
        MinimumStringLengthAfterRemovingSubstrings m = new MinimumStringLengthAfterRemovingSubstrings();
        System.out.println(m.minLength("ABFCACDB"));
    }

    
}

File: leetcode_2696/MinimumStringLengthAfterRemovingSubstrings.py
class MinimumStringLengthAfterRemovingSubstrings:
    def min_length(self, s: str) -> int:
        stack = []
        for c in s:
            if not stack:
                stack.append(c)
                continue
            if c == 'B' and stack[-1] == 'A':
                stack.pop()
            elif c == 'D' and stack[-1] == 'C':
                stack.pop()
            else:
                stack.append(c)
        return len(stack)


def main():
    main_instance = MinimumStringLengthAfterRemovingSubstrings()
    print(main_instance.min_length("ABFCACDB"))


if __name__ == "__main__":
    main()
